File: February 23, 2026.js
function canDonate(donor, recipient) {

    if (donor.includes("+") && recipient.includes("-")) {
        return false;
    }

    if (donor.includes("AB") && !recipient.includes("AB")) {
        return false;
    }

    if (donor.includes("B") && !(recipient.includes("B") || recipient.includes("AB"))) {
        return false;
    }

    if (donor.includes("A") && !(recipient.includes("A") || recipient.includes("AB"))) {
        return false;
    }

    return true;
}

File: February 23, 2026.py
def can_donate(donor, recipient):
    if ("+" in donor) and ("-" in recipient):
        return False
    
    if ("AB" in donor) and not("AB" in recipient):
        return False
    
    if ("B" in donor) and not(("B" in recipient) or ("AB" in recipient)):
        return False

    if ("A" in donor) and not(("A" in recipient) or ("AB" in recipient)):
        return False
    
    return True